### PR TITLE
Improve logs to include astronomer-cosmos identifier

### PR DESCRIPTION
**Pull request by tatiana**
Previously, we were using Airflow standard logging format for Cosmos messages, which looks like:
```
    [2023-08-09T14:20:55.532+0100] {subprocess.py:94} INFO - 13:20:55  Completed successfully
```

This change introduces a (yellow) "astronomer-cosmos" string into the project's log messages:
```
    [2023-08-09T14:20:55.532+0100] {subprocess.py:94} INFO - (astronomer-cosmos) - 13:20:55  Completed successfully
```

It also adds more logs to DBT project parsing to help troubleshoot issues while rendering the DAG.

Example of how the output looks like:
Screenshot 2023-08-09 at 15 24 56

Closes: #447

### CODE REVIEW

File: cosmos/log.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    Airflow logs usually look like:
+    [2023-08-09T14:20:55.532+0100] {subprocess.py:94} INFO - 13:20:55  Completed successfully
+
+    By using this logger, we introduce a (yellow) astronomer-cosmos string into the project's log messages:
+    [2023-08-09T14:20:55.532+0100] {subprocess.py:94} INFO - (astronomer-cosmos) - 13:20:55  Completed successfully
+    """
+    logger = logging.getLogger(name)
+    logger.propagate = False
+    formatter: logging.Formatter = CustomTTYColoredFormatter(fmt=LOG_FORMAT)  # type: ignore
```
Comment (tatiana) [inline cosmos/log.py:28]: The previous comment also applies to the class selection.

File: cosmos/dbt/graph.py [hunk 2]
```diff
@@ -194,10 +194,14 @@ def load_via_dbt_ls(self) -> None:
                     config=node_dict["config"],
                 )
                 nodes[node.unique_id] = node
+                logger.info("Parsed dbt resource `%s` of type `%s`", node.unique_id, node.resource_type)
```
Comment (harels) [inline cosmos/dbt/graph.py:197]: maybe this should be debug? sounds like this might be a bit too verbose?

Comment (tatiana) [reply cosmos/dbt/graph.py]: Done!

File: cosmos/hooks/subprocess.py [hunk 3]
```diff
@@ -77,7 +81,7 @@ def pre_exec() -> None:
                 preexec_fn=pre_exec,
             )
 
-            self.log.info("Output:")
+            logger.info("Output:")
```
Comment (harels) [inline cosmos/hooks/subprocess.py:84]:
```suggestion
            logger.info("Command output:")
```

File: cosmos/log.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+from airflow.utils.log.colored_log import CustomTTYColoredFormatter
+
+
+LOG_FORMAT: str = (
+    "[%(blue)s%(asctime)s%(reset)s] "
+    "{%(blue)s%(filename)s:%(reset)s%(lineno)d} "
+    "%(log_color)s%(levelname)s%(reset)s - "
+    "%(yellow)s(astronomer-cosmos)%(reset)s - "
+    "%(log_color)s%(message)s%(reset)s"
+)
```
[the remark this answers is not in the document]
Comment (harels) [reply cosmos/log.py:13]: I think this is good

File: dev/dags/dbt/jaffle_shop/profiles_local.yml
Comment (harels) [inline dev/dags/dbt/jaffle_shop/profiles_local.yml:1]: this seems unrelated to this pr?

File: tests/test_log.py
Comment (harels) [inline tests/test_log.py:1]: thanks for adding tests!

File: cosmos/log.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (tatiana) [reply cosmos/log.py:13]: Thanks, @harels, let's improve over time!
